docs/book/provider_implementations/generate_crds.md

```diff
@@ -0,0 +1,64 @@
+
+{% method %}
+### Create a repository
+
+{% sample lang="bash" %}
+```bash
+mkdir ${GOPATH}/src/sigs.k8s.io/cluster-api-provider-solas
+cd ${GOPATH}/src/sigs.k8s.io/cluster-api-provider-solas
+git init
+```
+{% endmethod %}
+
+{% method %}
+### Generate scaffolding
+
+When asked:
+
+```
+Run `dep ensure` to fetch dependencies (Recommended) [y/n]?
+y
+```
+
+you should.
+
+{% sample lang="bash" %}
+```bash
+kubebuilder init --domain cluster.k8s.io --license apache2 --owner "The Kubernetes Authors"
+```
+```bash
+git add .
+git commit -m "Generate scaffolding."
+```
+{% endmethod %}
+
+{% method %}
+### Generate provider resources for Clusters and Machines
+
+Here you will be asked if you want to generate resources and controllers.
+You only want to generate the resources because the controllers are defined
+in the common Cluster API code. Once we define Actuators in the next section
+we will register them with the existing Cluster API controllers.
+
+```
+Create Resource under pkg/apis [y/n]?
+y
+Create Controller under pkg/controller [y/n]?
+n
+```
+
+{% sample lang="bash" %}
+```bash
+kubebuilder create api --group solas --version v1alpha1 --kind SolasClusterProviderSpec
+kubebuilder create api --group solas --version v1alpha1 --kind SolasClusterProviderStatus
+```
+```bash
+kubebuilder create api --group solas --version v1alpha1 --kind SolasMachineProviderSpec
+kubebuilder create api --group solas --version v1alpha1 --kind SolasMachineProviderStatus
+```
+
```

Comment:
Should there be additional `kubebuilder create` commands to create the controllers and the manager scaffolding as well?

Reply:
I don't think so. The manager is created automatically, and `kubebuilder create` will prompt the user for whether they would like controllers to be generated as well. Since we reuse the Cluster API controllers, we only need to register the existing controllers with the manager.

Which reminds me: I am not sure if we ever define what a manager is...